google/cloud/pubsub/internal/publisher_tracing_connection_test.cc

```diff
@@ -135,6 +136,35 @@ TEST(PublisherTracingConnectionTest, PublishSpanOnError) {
                         "messaging.message.total_size_bytes", 45)))));
 }
 
+TEST(PublisherTracingConnectionTest, PublishSpanOmitsOrderingKey) {
+  namespace sc = ::opentelemetry::trace::SemanticConventions;
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<MockPublisherConnection>();
+  EXPECT_CALL(*mock, Publish)
+      .WillOnce([&](pubsub::PublisherConnection::PublishParams const&) {
+        EXPECT_TRUE(ThereIsAnActiveSpan());
+        return make_ready_future(StatusOr<std::string>("test-id-0"));
+      });
+  auto connection = MakePublisherTracingConnection(
+      Topic("test-project", "test-topic"), std::move(mock));
+
+  auto response = connection
+                      ->Publish({pubsub::MessageBuilder{}
+                                     .SetData("test-data-0")
+                                     .SetOrderingKey("")
+                                     .Build()})
+                      .get();
+
+  EXPECT_STATUS_OK(response);
+  EXPECT_THAT(span_catcher->GetSpans(),
+              ElementsAre(AllOf(
+                  SpanHasInstrumentationScope(), SpanKindIsProducer(),
+                  SpanNamed("projects/test-project/topics/test-topic send"),
+                  SpanWithStatus(opentelemetry::trace::StatusCode::kOk),
+                  Not(SpanHasAttributes(OTelAttribute<std::string>(
+                      "messaging.pubsub.ordering_key", ""))))));
```

Comment:
To be more general, I would `s/""/_/`

(and maybe you need a `using ::testing::_;` somewhere.

Reply:
Done